test/unit/spec/common/jobManager-Spec.js

```diff
@@ -971,338 +994,506 @@ define([
+                                this.jobManagerInstance.listeners[channelId][event]
+                            ).toBeDefined();
+                            TestUtil.sendBusMessage({
+                                bus: this.bus,
+                                message: error,
+                                channelType,
+                                channelId,
+                                type: event,
+                            });
+                        }).then(() => {
+                            expect(this.bus.emit).not.toHaveBeenCalled();
+                            expect(this.jobManagerInstance.updateModel).not.toHaveBeenCalled();
+                        });
                     });
                 });
             });
 
             describe('handleJobStatus', () => {
                 const event = jcm.MESSAGE_TYPE.STATUS;
-                it('can update the model if a job has been updated', function () {
-                    // job to test
+
+                // TODO: add error updates!!
+                describe('single job, multiple updates', () => {
                     const jobState = TestUtil.JSONcopy(JobsData.allJobsWithBatchParent)[0],
-                        jobId = jobState.job_id,
-                        channelType = jcm.CHANNEL.JOB,
-                        channelId = jobId,
-                        messageAddress = {
-                            type: event,
-                            channelType,
-                            channelId,
-                        },
-                        channelData = { channelType, channelId };
+                        jobId = jobState.job_id;
 
-                    this.jobId = jobId;
-                    setUpHandlerTest(this, event);
-                    Jobs.populateModelFromJobArray(
-                        this.jobManagerInstance.model,
-                        JobsData.allJobsWithBatchParent
-                    );
-                    // create an update for the job
-                    const updateOne = Object.assign({}, jobState, {
-                            updated: jobState.updated + 10,
-                        }),
-                        updateTwo = Object.assign({}, jobState, {
-                            retry_ids: [1, 2, 3],
-                        }),
-                        messageOne = { [jcm.PARAM.JOB_ID]: jobId, jobState },
-                        messageTwo = { [jcm.PARAM.JOB_ID]: jobId, jobState: updateOne },
-                        messageThree = { [jcm.PARAM.JOB_ID]: jobId, jobState: updateTwo };
+                    const inputs = {
+                        'by job ID': {
+                            channelType: jcm.CHANNEL.JOB,
+                            channelId: jobId,
+                        },
+                        'by batch ID': {
+                            channelType: jcm.CHANNEL.BATCH,
+                            channelId: JobsData.batchJob.batchId,
+                        },
```

Comment:
I got lazy here and just tested a single job addressed by job ID and a set of jobs addressed by batch ID